Travel Scheduler/View Controllers/FirstScreenViewController.m

```diff
@@ -88,6 +101,61 @@ - (void)viewDidLoad {
     [self.view addSubview:self.placesSearchBar];
     [self createLabels];
     [self createButton];
+    [self createFilterForGMSAutocomplete];
+    [self createAutocompleteTableView];
+    self.autocompleteTableView.delegate = self;
+    self.autocompleteTableView.dataSource = self;
+}
+
+#pragma mark - GMSAutocompleteFetcherDelegate
+
+NSMutableArray *resultsArr;
+- (void)didAutocompleteWithPredictions:(NSArray *)predictions
+{
+    resultsArr = [[NSMutableArray alloc] init];
+    for (GMSAutocompletePrediction *prediction in predictions) {
+        [resultsArr addObject:[prediction.attributedFullText string]];
+    }
+}
+
+- (void)didFailAutocompleteWithError:(NSError *)error
+{
+    NSString *errorMessage = [NSString stringWithFormat:@"%@", error.localizedDescription];
+    NSLog(@"%@", errorMessage);
+}
+
+#pragma mark - Autocomplete Delegate & TabeView DataSource Method
+
+- (nonnull UITableViewCell *)tableView:(nonnull UITableView *)tableView cellForRowAtIndexPath:(nonnull NSIndexPath *)indexPath
+{
+    static NSString *cellIdentifier = @"AutocompleteTableViewCell";
+    AutocompleteTableViewCell *cell = (AutocompleteTableViewCell *)[tableView dequeueReusableCellWithIdentifier:cellIdentifier];
+    if (cell == nil) {
+        cell = [[AutocompleteTableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:cellIdentifier];
+        cell.backgroundColor=[UIColor clearColor];
+        cell.textLabel.textColor=[UIColor blackColor];
+    }
+    cell.textLabel.text = [resultsArr objectAtIndex:indexPath.row];
+    return cell;
+}
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
+{
+    return 1;
+}
+
+- (NSInteger)tableView:(nonnull UITableView *)tableView numberOfRowsInSection:(NSInteger)section
+{
+    return resultsArr.count;
```

Comment:
How big can this be? Should we check to make sure this isn't too big or are there API-side checks for that?

Reply:
Since my filter is mainly places and all roads and other streets are filtered out, I think we should not be concerned. Also, the filter considers bounds given. I currently set it to be centered at Australia but I can change that to SF and also this would be helpful in the different autocomplete I am currently working on.